File: PetShop_c#/Animal.cs
using System;

public abstract class Animal
{
    public abstract double GetPrix();
    public abstract string GetAnimalType();
}

public class Chien : Animal
{
    public override double GetPrix()
    {
        return 100;
    }

    public override string GetAnimalType()
    {
        return "Chien";
    }
}

public class Chat : Animal
{
    public override double GetPrix()
    {
        return 110;
    }

    public override string GetAnimalType()
    {
        return "Chat";
    }
}

File: PetShop_Java/src/Animal.java
public abstract class Animal {
    //---Variables
    protected double prix;
    protected String type;


    public Animal(double prix, String type) {
        this.prix = prix;
        this.type = type;
    }
    public abstract double getPrix();
    public abstract String getType();
}

//---Classe Chien---
class Chien extends Animal {
    public Chien() {
        super(100, "Chien");
    }

    @Override
    public double getPrix() {
        return prix;
    }

    @Override
    public String getType() {
        return type;
    }
}

//---Classe Chat
class Chat extends Animal {
    public Chat() {
        super(110, "Chat");
    }

    @Override
    public double getPrix() {
        return prix;
    }

    @Override
    public String getType() {
        return type;
    }
}
